client/src/main/kotlin/tech/figure/classification/asset/client/client/impl/DefaultACQuerier.kt

```diff
@@ -44,78 +53,200 @@ class DefaultACQuerier(
+        ),
+        throwExceptions = throwExceptions,
+    )
 
-    override fun queryContractVersion(): ACVersionInfo = doQuery(QueryVersion)
+    override fun queryFeePaymentsByAssetUuid(
+        assetUuid: UUID,
+        assetType: String,
+    ): FeePaymentDetail = doQuery(
+        query = QueryFeePayments(
+            identifier = AssetIdentifier.AssetUuid(value = assetUuid),
+            assetType = assetType,
+        )
+    )
+
+    override fun queryFeePaymentsByScopeAddressOrNull(
+        scopeAddress: String,
+        assetType: String,
+        throwExceptions: Boolean,
+    ): FeePaymentDetail? = doQueryOrNull(
+        query = QueryFeePayments(
+            identifier = AssetIdentifier.ScopeAddress(value = scopeAddress),
+            assetType = assetType,
+        ),
+        throwExceptions = throwExceptions,
+    )
+
+    override fun queryFeePaymentsByScopeAddress(
+        scopeAddress: String,
+        assetType: String,
+    ): FeePaymentDetail = doQuery(
+        query = QueryFeePayments(
+            identifier = AssetIdentifier.ScopeAddress(value = scopeAddress),
+            assetType = assetType,
+        )
+    )
+
+    override fun queryContractState(): ACContractState = doQuery(
+        query = QueryState,
+    )
+
+    override fun queryContractVersion(): ACVersionInfo = doQuery(
+        query = QueryVersion,
+    )
 
     /**
      * Executes a provided [ContractQuery] against the Asset Classification smart contract.  This relies on the
      * internalized [PbClient] to do the heavy lifting.
      */
-    private inline fun <reified T : ContractQuery, reified U : Any> doQuery(query: T): U =
-        doQueryOrNull(query)
-            ?: throw NullContractResponseException("Received null response from asset classification smart contract for: ${query.queryFailureMessage}")
+    private inline fun <reified T : ContractQuery, reified U : Any> doQuery(
+        query: T,
+        typeReference: TypeReference<U>? = null,
+    ): U = doQueryOrNull(query = query, typeReference = typeReference)
+        ?: throw NullContractResponseException("Received null response from asset classification smart contract for: ${query.queryFailureMessage}")
 
-    private inline fun <reified T : ContractQuery, reified U : Any> doQueryOrNull(query: T): U? =
+    private inline fun <reified T : ContractQuery, reified U : Any> doQueryOrNull(
+        query: T,
+        typeReference: TypeReference<U>? = null,
+    ): U? =
         pbClient.wasmClient.queryWasm(
             QueryOuterClass.QuerySmartContractStateRequest.newBuilder()
                 .setAddress(queryContractAddress())
                 .setQueryData(query.toBase64Msg(objectMapper))
                 .build()
         ).data
             .toByteArray()
-            .let { array -> objectMapper.readValue(array, U::class.java) }
+            .let { array ->
+                if (typeReference != null) {
+                    objectMapper.readValue(array, typeReference)
```

Comment:
nit: `typeReference :? U::class.java` instead?

Reply:
RIP cain't do it.  Type ref and java class are two separate overloads for `readValue` so it has to be separate invocations like this to compile because kotlin isn't magic enough to dynamically infer overload choice from provided types

Reply:
All that said, I could do something like:
```
typeReference?.let { objectMapper.readValue(array, it) } ?: objectMapper.readValue(array, U::class.java)
```

Reply:
oh, got it, yeah nevermind then